packages/react-data-grid/src/utils/SelectedCellUtils.ts

```diff
@@ -0,0 +1,183 @@
+import { CellNavigationMode, Z_INDEXES } from '../common/enums';
+import * as rowUtils from '../RowUtils';
+import { getColumn, isFrozen, canEdit } from '../ColumnUtils';
+import { Column, Position, Range, Dimension, RowGetter } from '../common/types';
+
+const getRowTop = (rowIdx: number, rowHeight: number): number => rowIdx * rowHeight;
+
+interface getSelectedRowOpts {
+  selectedPosition: Position;
+  rowGetter: RowGetter;
+}
+
+export function getSelectedRow({ selectedPosition, rowGetter }: getSelectedRowOpts) {
+  return rowGetter(selectedPosition.rowIdx);
+}
+
+interface getSelectedDimensionsOpts {
+  selectedPosition: Position;
+  columns: Column[];
+  rowHeight: number;
+  scrollLeft: number;
+}
+
+export function getSelectedDimensions({ selectedPosition: { idx, rowIdx }, columns, rowHeight, scrollLeft }: getSelectedDimensionsOpts): Dimension {
+  if (idx < 0) {
+    return { width: 0, left: 0, top: 0, height: rowHeight, zIndex: 1 };
+  }
+  const column = getColumn(columns, idx);
+  const frozen = isFrozen(column);
+  const { width } = column;
+  const left = frozen ? column.left + scrollLeft : column.left;
+  const top = getRowTop(rowIdx, rowHeight);
+  const zIndex = frozen ? Z_INDEXES.FROZEN_CELL_MASK : Z_INDEXES.CELL_MASK;
+  return { width, left, top, height: rowHeight, zIndex };
+}
+
+interface getSelectedRangeDimensionsOpts {
+  selectedRange: Range;
+  columns: Column[];
+  rowHeight: number;
+}
+
+export function getSelectedRangeDimensions({ selectedRange: { topLeft, bottomRight }, columns, rowHeight }: getSelectedRangeDimensionsOpts): Dimension {
+  if (topLeft.idx < 0) {
+    return { width: 0, left: 0, top: 0, height: rowHeight, zIndex: Z_INDEXES.CELL_MASK };
+  }
+
+  let width = 0;
+  let anyColFrozen = false;
+  for (let i = topLeft.idx; i <= bottomRight.idx; i++) {
+    const column = getColumn(columns, i);
+    width += column.width;
+    anyColFrozen = anyColFrozen || isFrozen(column);
+  }
+
+  const { left } = getColumn(columns, topLeft.idx);
+  const top = getRowTop(topLeft.rowIdx, rowHeight);
+  const height = (bottomRight.rowIdx - topLeft.rowIdx + 1) * rowHeight;
+  const zIndex = anyColFrozen ? Z_INDEXES.FROZEN_CELL_MASK : Z_INDEXES.CELL_MASK;
+
+  return { width, left, top, height, zIndex };
+}
+
+interface getSelectedColumnOpts {
+  selectedPosition: Position;
+  columns: Column[];
+}
+
+export function getSelectedColumn({ selectedPosition, columns }: getSelectedColumnOpts): Column {
+  return getColumn(columns, selectedPosition.idx);
+}
+
+interface getSelectedCellValueOpts {
+  selectedPosition: Position;
+  columns: Column[];
+  rowGetter: RowGetter;
+}
+
+export function getSelectedCellValue({ selectedPosition, columns, rowGetter }: getSelectedCellValueOpts) {
```

Comment:
consider adding return type?

Reply:
I need to tsify rowutils first, otherwise I don''t know yet what the return type should be. Should I add a TODO comment?